src/main/java/org/zalando/problem/spring/web/advice/validation/ConstraintViolationProblem.java

```diff
@@ -13,30 +10,28 @@
 import java.util.List;
 
 @Immutable
-@JsonTypeName(ConstraintViolationProblem.TYPE_VALUE)
 public final class ConstraintViolationProblem extends ThrowableProblem {
 
     public static final String TYPE_VALUE = "https://zalando.github.io/problem/constraint-violation";
     public static final URI TYPE = URI.create(TYPE_VALUE);
 
+    private final URI type;
     private final StatusType status;
-    private final String detail;
     private final List<Violation> violations;
 
     public ConstraintViolationProblem(final StatusType status, final List<Violation> violations) {
-        this(status, null, new ArrayList<>(violations));
+        this(TYPE, status, new ArrayList<>(violations));
     }
 
-    @JsonCreator
-    ConstraintViolationProblem(final StatusType status, @Nullable final String detail, final List<Violation> violations) {
+    ConstraintViolationProblem(final URI type, final StatusType status, final List<Violation> violations) {
```

Comment:
So ne JsonCreator anymore? It would also be impossible to provide detail with this change.